build.go

```diff
@@ -51,6 +51,9 @@ type flattenedField struct {
 
 func flattenedFields(v reflect.Value, ptag *Tag) (out []flattenedField, err error) {
 	v = reflect.Indirect(v)
+	if v.Kind() != reflect.Struct {
```

Comment:
I like the general idea, but I think this should be a bit more selective about what it allows, because AFAICT this will now allow any value at all? I think adding an explicit check, probably in `DynamicCommand`, for allowed types, would be good.

Reply:
yes, good point; AFAICT the only thing we really require is the "not-quite-interface" of having some `Run(...)` method, right?

Reply:
added a first attempt; PTAL :pray:

Reply:
@alecthomas ping? :innocent: